### PR TITLE
Gives the Bartender/Chef the Ability to Jump onto Tables Faster

### PR DESCRIPTION
**Pull request by matttheficus**
## What Does This PR Do
Allows the Barkeep/Chef to almost instantly jump up onto tables via the `TRAIT_TABLE_LEAP`

## Why It's Good For The Game
You can now leap over the Bar (and other tables) to go smack nerds who are disrupting the Bar. Nothing funnier than the Barkeep going "HEY!" and leaping over the Bar to put stamshot in your ass.

Also give to the Chef so they can do the same deal for kitchen trespassers who loop around any tables they set up in the kitchen area.

## Testing
Jumped up on some tables.

## Changelog
:cl:
add: Added `TRAIT_TABLE_LEAP` to allow for Chef/Bartender to jump up onto tables faster.
/:cl:

### CODE REVIEW

File: code/game/objects/structures.dm [hunk 1]
```diff
@@ -73,9 +73,16 @@
 
 	usr.visible_message("<span class='warning'>[user] starts climbing onto \the [src]!</span>")
 	climber = user
-	if(!do_after(user, 50, target = src))
-		climber = null
-		return FALSE
+	if(climber.mind.assigned_role == "Bartender" || climber.mind.assigned_role == "Chef")
```
Comment (Contrabang) [inline code/game/objects/structures.dm:76]: Could you make this a trait instead that is added in `/datum/outfit/job/bartender/post_equip`, with the source being a new trait source called `JOB_TRAIT` (this is what I'd prefer), if you're lazy, `ROUNDSTART_TRAIT` works as well

File: code/game/objects/structures.dm [hunk 2]
```diff
@@ -73,9 +73,16 @@
...
+		usr.visible_message("<span class='warning'>[user] gets ready to vault up onto \the [src]!</span>")
```
Comment (lewcc) [inline code/game/objects/structures.dm:77]:
```suggestion
		user.visible_message("<span class='warning'>[user] gets ready to vault up onto [src]!</span>")
```

File: code/game/objects/structures.dm [hunk 3]
```diff
@@ -73,9 +73,16 @@
...
+		if(!do_after(user, 5, target = src))
+			climber = null
+			return FALSE
+	else
+		usr.visible_message("<span class='warning'>[user] starts climbing onto \the [src]!</span>")
```
Comment (lewcc) [inline code/game/objects/structures.dm:82]:
```suggestion
		user.visible_message("<span class='warning'>[user] starts climbing onto [src]!</span>")
```

File: code/game/objects/structures.dm [hunk 4]
```diff
@@ -73,9 +73,16 @@
...
+		if(!do_after(user, 50, target = src))
```
Comment (lewcc) [inline code/game/objects/structures.dm:83]:
```suggestion
		if(!do_after(user, 5 SECONDS, target = src))
```

File: code/game/objects/structures.dm [hunk 5]
```diff
@@ -73,9 +73,16 @@
 
 	usr.visible_message("<span class='warning'>[user] starts climbing onto \the [src]!</span>")
 	climber = user
-	if(!do_after(user, 50, target = src))
-		climber = null
-		return FALSE
+	if(climber.mind.assigned_role == "Bartender" || climber.mind.assigned_role == "Chef")
+		usr.visible_message("<span class='warning'>[user] gets ready to vault up onto \the [src]!</span>")
+		if(!do_after(user, 5, target = src))
```
Comment (lewcc) [inline code/game/objects/structures.dm:78]:
```suggestion
		if(!do_after(user, 0.5 SECONDS, target = src))
```

File: code/game/objects/structures.dm [hunk 6]
```diff
@@ -73,9 +73,16 @@
 
 	usr.visible_message("<span class='warning'>[user] starts climbing onto \the [src]!</span>")
```
Comment (lewcc) [inline code/game/objects/structures.dm:74]: This message will be duplicated, you probably want to remove it from here

**Comment by SpringSkipper** [comment]
The new time is 0.5 seconds???
that's a **huge** advantage in a lot of areas other than just the bar and kitchen. I won't argue that the problem is two jobs that don't have a lot of spicy antag-useful item access getting a bit stronger that way. But I also don't like the idea of starting to pass out job-specific advantages to crew based on mind roles in a way that makes job swapping at the HoP line harder, identifying clings easier, and otherwise goes against the "anyone can do anything with the right tools and knowledge" principles that ss13 generally goes by.

So far as I know mime, chef, and clown are the only three jobs that have something 'special' about them at roundstart tied to their mind role.

File: code/game/objects/structures.dm (hunk 1 above)
Comment (matttheficus) [reply code/game/objects/structures.dm:76]: Given I only want this on roundstart/latejoin jobs, i think ill just toss it on `ROUNDSTART_TRAIT`

**Comment by matttheficus** [comment]
Mime, Chef, Detective, Bartender, Clown, Chief Engineer, and the Chaplain all have slightly different roundstart "things" simply by existing. You COULD technically sober-check the maybe fake Detective, but that's one of those things where if people are doing those kinda checks, theyre likely not playing the game for the betterment or fun of the community anymore.

I also dont think this diverges from any design ideal that SS13 has. I dont see the game as something where "anybody can do everything if they go and get the tools/knowledge" I see it as a game that has so many different facets and neat little intricacies where you use the tools and items of your Job and Role to work toward a goal, whether that be RP or Antagging or whatever you so choose.

**Comment by Adrer** [comment]
I like the idea but I feel like going from 5 seconds to .5 seconds is a bit TOO much; then again I'd have to play around with it to see what would be a more reasonable time.

**Comment by datlo** [comment]
Alternative way that avoids giving the jobs inherent power while getting what you want : Maybe this could be a genetic power that they get roundstart similar to sober/clumsy/etc?

**Review by S34NW — APPROVED** [review]

**Pull request merged**